src/lib/server/server-provider.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {
+  InjectionToken,    // tslint:disable-line:no-unused-variable
+  ComponentRef,      // tslint:disable-line:no-unused-variable
+} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+import {BEFORE_APP_SERIALIZED} from '@angular/platform-server';
+
+import {
+  BreakPoint,
+  BREAKPOINTS,
+  CLASS_NAME,
+  MatchMedia,
+  ServerStylesheet,
+  SERVER_TOKEN,
+} from '@angular/flex-layout';
+
+let UNIQUE_CLASS = 0;
+const DEBUG_FLAG = false;
+
+/**
+ * create @media queries based on a virtual stylesheet
+ * * Adds a unique class to each element and stores it
+ *   in a shared classMap for later reuse
+ */
+function formatStyle(stylesheet: Map<HTMLElement, Map<string, string|number>>,
+                     _document: Document,
+                     mediaQuery: string,
+                     classMap: Map<HTMLElement, string>) {
+  let styleText = DEBUG_FLAG ? `
+        @media ${mediaQuery} {` : `@media ${mediaQuery}{`;
+  stylesheet.forEach((styles, el) => {
+    let className = classMap.get(el);
+    if (!className) {
+      className = `${CLASS_NAME}${UNIQUE_CLASS++}`;
+      classMap.set(el, className);
+    }
+    el.classList.add(className);
+    styleText += DEBUG_FLAG ? `
+          .${className} {` : `.${className}{`;
+    styles.forEach((v, k) => {
+      if (v) {
+        styleText += DEBUG_FLAG ? `
+              ${k}: ${v};` : `${k}:${v};`;
+      }
+    });
+    styleText += DEBUG_FLAG ? `
+          }` : '}';
+  });
+  styleText += DEBUG_FLAG ? `
+        }\n` : '}';
+
+  return styleText;
+}
+
+/**
+ * Add or remove static styles depending on the current
+ * platform
+ * format the static @media queries for all breakpoints
+ * to be used on the server and append them to the <head>
+ */
+export function addStyles(serverSheet: ServerStylesheet,
```

Comment:
`addStyles` is not very specific. Could the name be something more like `generateStaticFlexLayoutStyles`? It would also be good if the description went into more detail about how it works.

Reply:
Added